src/models/base.es6.js

```diff
@@ -62,62 +44,50 @@ class Base {
       return true;
     }
 
-    this.emit('validationError', invalid);
     return invalid;
   }
 
-  format (prop, value){
-    if (!this.formatters || !this.formatters[prop]) {
-      return value;
-    }
-
-    return this.formatters[prop](value);
-  }
-
-  toJSON (formatter) {
-    if (formatter && typeof formatter == 'function') {
-      return formatter(this.props);
-    }
+  toJSON () {
+    const type = this._type;
 
-    return this.props;
+    return Object.assign({}, this.props, {
+      _type: type,
+    });
   }
-}
-
-Base.validators = {
-  integer: function(i) {
-    return i === parseInt(i);
-  },
 
-  string: function(s) {
-    return s === s.toString();
-  },
+  static validators = {
+    integer: function(i) {
+      return i === parseInt(i);
+    },
 
-  min: function (i, min) {
-    return i >= min;
-  },
+    string: function(s) {
```

Comment:
:no_bell: That's odd, I wonder if its our babel config/version. I use [babel's online repl](https://www.babeljs.io/repl) to experiment a lot and the following compiles just fine:
```
class ApiModel {
  static validators = {
    string(s) {
      return s === s.toString()
    }, 
    
    minLength: function(i, min) {
      return i >= min;
    },
    
    maxLength(i, max) {
      return i <= max;
    },
  }
}
```